lib/modules/dosomething/dosomething_user/dosomething_user.module

```diff
@@ -698,6 +699,11 @@ function dosomething_user_user_insert(&$edit, &$account, $category = NULL) {
     $edit['name'] = $account->uid;
     $account->name = $account->uid;
   }
+
+  // Trigger an analytics event on registration
+  // @NOTE: The 'Authentication - Login' event will *also* be triggered here,
+  // since registered users are subsequently logged in to their new accounts.
+  dosomething_helpers_add_analytics_event('Authentication', 'Register');
```

Comment:
Hmm, I looked at the `new_user` method but it seemed to be mostly concerned with transactional logic. What's the benefit of doing things in that method over the `user_insert` hook?